src/markov/mc_tools.jl

```diff
@@ -224,6 +224,46 @@ function mc_compute_stationary{T}(mc::MarkovChain{T})
     return stationary_dists
 end
 
+
+for (S, ex) in ((Real, :(T)), (Integer, :(Rational{T})))
+    @eval function stationary_distributions{T<:$S}(mc::MarkovChain{T})
+        n = n_states(mc)
+        rec_classes = recurrent_classes(mc)
+        T1 = $ex
+        stationary_dists = Array(Vector{T1}, length(rec_classes))
+
+        if length(rec_classes) == 1  # unique recurrent class
+            stationary_dists[1] = gth_solve(mc.p)
+        else  # more than one recurrent classes
+            for i in 1:length(rec_classes)
+                rec_class = rec_classes[i]
+                dist = zeros(T1, n)
+                dist[rec_class] = gth_solve(sub(mc.p, rec_class, rec_class))
```

Comment:
I thought use of `sub` would be a good idea, but in the case of a sparse `mc.p`, the `full` in `gth_solve` does not change `SubArray{SparseMatrixCSC}` to dense. Maybe I should extend `full` to `SubArray{SparseMatrixCSC}`?